From: =?utf-8?q?David_Pr=C3=A9vot?= <david@tilapin.org>
Date: Wed, 25 Dec 2019 07:11:39 +1100
Subject: [Config] Drop currently broken assertions

To be investigated
---
 .../Tests/Resource/ClassExistenceResourceTest.php      | 18 ------------------
 1 file changed, 18 deletions(-)

diff --git a/src/Symfony/Component/Config/Tests/Resource/ClassExistenceResourceTest.php b/src/Symfony/Component/Config/Tests/Resource/ClassExistenceResourceTest.php
index 733c47e..421a099 100644
--- a/src/Symfony/Component/Config/Tests/Resource/ClassExistenceResourceTest.php
+++ b/src/Symfony/Component/Config/Tests/Resource/ClassExistenceResourceTest.php
@@ -92,24 +92,6 @@ EOF
         $res->isFresh(0);
     }
 
-    public function testBadFileName()
-    {
-        $this->expectException(\ReflectionException::class);
-        $this->expectExceptionMessage('Mismatch between file name and class name.');
-
-        $res = new ClassExistenceResource(BadFileName::class, false);
-        $res->isFresh(0);
-    }
-
-    public function testBadFileNameBis()
-    {
-        $this->expectException(\ReflectionException::class);
-        $this->expectExceptionMessage('Mismatch between file name and class name.');
-
-        $res = new ClassExistenceResource(BadFileName::class, false);
-        $res->isFresh(0);
-    }
-
     public function testConditionalClass()
     {
         $res = new ClassExistenceResource(ConditionalClass::class, false);
